### PR TITLE
Exposed BitArray methods as public and fixed tests

### PR DESCRIPTION
**Pull request by Lixire**
- Changed methods to be public
- Fixed CopyTo_Type_Invalid() tests

Regarding #9994

### CODE REVIEW

**Comment by Lixire** [comment]
@danmosemsft

File: src/System.Collections/tests/BitArray/BitArray_CtorTests.cs [hunk 1]
```diff
@@ -31,9 +31,8 @@ public static void Ctor_Int(int length)
                 Assert.False(bitArray[i]);
                 Assert.False(bitArray.Get(i));
             }
-            ICollection collection = bitArray;
-            Assert.Equal(length, collection.Count);
-            Assert.False(collection.IsSynchronized);
+            Assert.Equal(length, bitArray.Count);
+            Assert.False(bitArray.IsSynchronized);
```
Comment (danmosemsft) [inline src/System.Collections/tests/BitArray/BitArray_CtorTests.cs]: >Assert [](start = 12, length = 6)

also add a line to test that bitArray.SyncRoot is not null, and if you get it twice, it's the same thing each time. no need to repeat below

**Comment by danmosemsft** [comment]
:shipit:

**Comment by Lixire** [comment]
@danmosemsft 
For the CopyTo_Type_Invalid() tests in BitArray_GetSetTests.cs , the dev/api branch currently differs from master (i.e., master has "array" which is what I have, instead of null). Is this why the builds are failing?

**Comment by danmosemsft** [comment]
You can merge this PR as soon as the CI is clean.

**Comment by Lixire** [comment]
Cool, thanks

**Comment by danmosemsft** [comment]
Master should have merged into dev/api today. Fetch latest?
________________________________

**Comment by danmosemsft** [comment]
The single failure is a known issue https://github.com/dotnet/coreclr/issues/6329

**Pull request merged**